src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java

```diff
@@ -101,6 +101,23 @@ public void execute() {
         openFiles(filesToOpen, true);
     }
 
+    /**
+     *
+     * @return Path of current panel database directory or the working directory
+     */
+    private Path getInitialDirectory() {
+        if (frame.getBasePanelCount() == 0) {
+            return getWorkingDirectoryPath();
+        } else {
+            Optional<Path> databasePath = frame.getCurrentBasePanel().getBibDatabaseContext().getDatabasePath();
+            return !databasePath.isPresent() ? getWorkingDirectoryPath() : databasePath.get().getParent();
```

Comment:
Thank you! The point is that if the value is present I don't want to return the value itself, but the .getParent(). And, int the other branch, I can not apply .getParent() to the value saved in the WORKING_DIRECTORY. I don't have lot of experience with optionals and don't know how to use the orElse in this situation.

Reply:
This should do the trick: `databasePath.map(p -> p.getParent()).orElse(getWorkingDirectoryPath())`

Reply:
Cool! I was not aware map() could be used with optionals! Thank you.

Reply:
Hi,

that should be possible with Optional.map:  (You can chain optionals similar to streams, here is a nice overview: https://www.nurkiewicz.com/2013/08/optional-in-java-8-cheat-sheet.html )

This will transform the value by executing getParent if the value is present, otherwise it would return the workingDir path.

`return databasePath.map(Path::getParent).orElse(getWorkingDirectoryPath?)`

PS: Sorry that I reply today, I saw that I did a review using githubs feature, but did not submit the review 🤦‍♂️ )

Reply:
Thank you again, @Siedlerchr. I just applied and pushed the refactor proposed by @koppor before seeing your comment.

Reply:
It's essentially the same, but using method references looks nicer ;)